src/game/Entities/Player.cpp

```diff
@@ -5882,22 +5882,46 @@ int32 Player::CalculateReputationGain(ReputationSource source, int32 rep, int32
 
     percent += rep > 0 ? repMod : -repMod;
 
+    uint32 plLevel = getLevel();
     float rate;
+    // Declines with 20% (min 0.2) for each level if 6 levels or more below the player
+    if (creatureOrQuestLevel < plLevel)
+    {
+        if (plLevel <= (creatureOrQuestLevel + 5))
+            rate = 1.0f;
+        else if (plLevel == (creatureOrQuestLevel + 6))
+            rate = 0.8f;
+        else if (plLevel == (creatureOrQuestLevel + 7))
+            rate = 0.6f;
+        else if (plLevel == (creatureOrQuestLevel + 8))
+            rate = 0.4f;
+        else
+            rate = 0.2f;
+    }
+
     switch (source)
     {
         case REPUTATION_SOURCE_KILL:
-            rate = sWorld.getConfig(CONFIG_FLOAT_RATE_REPUTATION_LOWLEVEL_KILL);
+        {
+            float configRate = sWorld.getConfig(CONFIG_FLOAT_RATE_REPUTATION_LOWLEVEL_KILL);
+            if ((rate < configRate) || rate > configRate && creatureOrQuestLevel <= MaNGOS::XP::GetGrayLevel(plLevel))
+                rate = configRate;
             break;
+        }
         case REPUTATION_SOURCE_QUEST:
-            rate = sWorld.getConfig(CONFIG_FLOAT_RATE_REPUTATION_LOWLEVEL_QUEST);
+        {
+            float configRate = sWorld.getConfig(CONFIG_FLOAT_RATE_REPUTATION_LOWLEVEL_QUEST);
+            if ((rate < configRate) || rate > configRate && creatureOrQuestLevel <= MaNGOS::XP::GetGrayLevel(plLevel))
+                rate = configRate;
             break;
+        }
         case REPUTATION_SOURCE_SPELL:
         default:
             rate = 1.0f;
             break;
     }
 
-    if (rate != 1.0f && creatureOrQuestLevel <= MaNGOS::XP::GetGrayLevel(getLevel()))
+    if (rate != 1.0f)
```

Comment:
What do you mean it is equalized, though? It's a float on float comparison, rate is a float not a uin32 and 1.0f is a float, where is the issue happening here? is it the != operation? Hmm..

Reply:
It should work as long as I am not doing a comparison against 1 or 1.0 but do it specifically against 1.0f like I am doing.

https://www.geeksforgeeks.org/comparison-float-value-c/

Reply:
Depends where the numbers come from I suppose. Generally you don't use equality or inequality comparisons on floating point numbers because rounding errors can make them be slightly off:

    >>> 0.3 - 0.2 == 0.1
    False

(This is from the python interpreter but they use the same IEEE754 floating point standard that intel CPUs and pretty much the whole world use)

However, if no operations are done on the floats then the comparisons will never cause issues, especially not on floats like 1.0 that have an exact representation.

TL;DR: I don't think this particular case will cause any issues.